toolkit/docs/how_it_works/0_intro.md

```diff
@@ -2,23 +2,308 @@ How The Build System Works
+        %% scheduler nodes
+        subgraph sched ["Scheduler tool (scheduler)"]
+        currentGraph[/Current Graph/]:::io
+        trim[Remove unneeded branches from graph]:::process
+        doneBuild{{Done building all required nodes?}}:::decision 
+        leafNodesAvail{{Leaf nodes available?}}:::decision
+        worker[Schedule a worker to build the SRPM]:::process
+        builtRPMs[/Built RPMs/]:::io
+        updateDeps[Update dependencies in graph]:::process
+        getDeps[Get dependencies]:::process
+
+        %% scheduler flow
+        currentGraph-->trim
+        trim-->doneBuild
+        doneBuild-->|no|leafNodesAvail
+        leafNodesAvail-->|yes|worker
+        builtRPMs-->updateDeps
+        updateDeps-->currentGraph
+        getDeps-->worker
+        worker-->builtRPMs
+
+        end
+    %% end of scheduler
+```
+
+<br/>
+
 ### Image Generation
 Image generation is discussed in detail [here](4_image_generation.md). The image generation step creates a new filesystem, and then installs packages into it based on the currently selected configuration file. These packages can be either the locally built packages, or packages pulled from one or more package servers. Once the filesystem is composited any additional changes listed in the config file are made, the filesystem is then packaged into the requested format (`vhd`, `vhdx`, `ISO`, etc).
 
+```mermaid
+flowchart TD
+    %% style definitions
+    classDef io fill:#247BA0,stroke:#333,stroke-width:2px,color:#fff;
+    classDef process fill:#B05E2F,stroke:#333,stroke-width:2px,color:#fff;
+    classDef decision fill:#51344D,stroke:#333,stroke-width:2px,color:#fff;
+    classDef goodState fill:#566E40,stroke:#333,stroke-width:2px,color:#fff;
+    classDef badState fill:#BC4B51,stroke:#333,stroke-width:2px,color:#fff;
+
+    %% state nodes
+    start(["Start (make image / make iso)"]):::goodState
+    done([Done]):::goodState
+    %% error[[Error]]:::badState
+
+    %% Image nodes
+    imageConfig[/Image config.json/]:::io
+    raw[/Raw image or file system/]:::io
+    roast["Image format converter (roast)"]:::process
+    initrd[/initrd/]:::io
+    iso[/"iso (imager tool, pkgs, config files)"/]:::io
+    image[/image/]:::io
+    pkgFetcher["Package fetcher (imagepkgfetcher)"]:::process
+    rpmCache[(local RPMs)]:::io
+    remoteRepo[(remote repo)]:::io
+    missingDep[/RPMs to fill Missing Dependencies/]:::io
+    imager["Image tool (imager)"]:::process
+    isoBuild{{iso installer or offline build?}}:::decision
+    isoBuilder["iso maker (isomaker)"]:::process
+
+    %% Image flow
+    start-->pkgFetcher
+    imageConfig-->pkgFetcher
+    rpmCache-->pkgFetcher
+    remoteRepo-->missingDep
+    missingDep-->pkgFetcher
+    pkgFetcher-->isoBuild
+    isoBuild-->|iso installer|isoBuilder
+    initrd-->isoBuilder
+    isoBuilder-->iso
+    iso-->done
+    isoBuild-->|offline|imager
+    imager-->raw
+    raw-->roast
+    roast-->image
+    image-->done
+    
+```
```

Comment:
```suggestion
```mermaid
flowchart TD
    %% style definitions
    classDef io fill:#247BA0,stroke:#333,stroke-width:2px,color:#fff;
    classDef process fill:#B05E2F,stroke:#333,stroke-width:2px,color:#fff;
    classDef decision fill:#51344D,stroke:#333,stroke-width:2px,color:#fff;
    classDef goodState fill:#566E40,stroke:#333,stroke-width:2px,color:#fff;
    classDef badState fill:#BC4B51,stroke:#333,stroke-width:2px,color:#fff;

    %% state nodes
    start(["Start (make image / make ISO)"]):::goodState
    done([Done]):::goodState
    %% error[[Error]]:::badState

    %% Image nodes
    imageConfig[/Image config.json/]:::io
    raw[/Raw image or file system/]:::io
    roast["Image format converter (roast)"]:::process
    initrd[/initrd/]:::io
    iso[/"ISO (imager tool, pkgs, config files)"/]:::io
    image[/image/]:::io
    pkgFetcher["Package fetcher (imagepkgfetcher)"]:::process
    rpmCache[(Local RPMs)]:::io
    remoteRepo[(Remote repo)]:::io
    missingDep[/RPMs to fill Missing Dependencies/]:::io
    imager["Image tool (imager)"]:::process
    isoBuild{{ISO installer or offline build?}}:::decision
    isoBuilder["ISO maker (isomaker)"]:::process

    %% Image flow
    start-->pkgFetcher
    imageConfig-->pkgFetcher
    rpmCache-->pkgFetcher
    remoteRepo-->missingDep
    missingDep-->pkgFetcher
    pkgFetcher-->isoBuild
    isoBuild-->|iso installer|isoBuilder
    initrd-->isoBuilder
    isoBuilder-->iso
    iso-->done
    isoBuild-->|offline|imager
    imager-->raw
    raw-->roast
    roast-->image
    image-->done
    
```
```